code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,685 @@
+			fish_num --
+			if(fish_num > 0)					//There's more fish, add a comma to the message
+				message +=", "
+		message +="."							//No more fish, end the message with a period
+		//Display the number of fish and previously constructed message
+		examine_message += "\The [src] contains [fish_list.len] live fish. [message] "
+
+	examine_message += "<br>"
+
+	//Report lid state for tanks and wall-tanks
+	if(has_lid)									//Only report if the tank actually has a lid
+		//Report lid state
+		if(lid_switch)
+			examine_message += "The lid is closed. "
+		else
+			examine_message += "The lid is open. "
+
+	examine_message += "<br>"
+
+	//Report if the tank is leaking/cracked
+	if(water_level > 0)							//Tank has water, so it's actually leaking
+		if(leaking == MINOR_LEAK)
+			examine_message += "\The [src] is leaking."
+		if(leaking == MAJOR_LEAK)
+			examine_message += "\The [src] is leaking profusely!"
+	else										//No water, report the cracks instead
+		if(leaking == MINOR_LEAK)
+			examine_message += "\The [src] is cracked."
+		if(leaking == MAJOR_LEAK)
+			examine_message += "\The [src] is nearly shattered!"
+
+
+	//Finally, report the full examine_message constructed from the above reports
+	to_chat(user, "[jointext(examine_message, "")]")
+	return examine_message
+
+//////////////////////////////
+//		ATACK PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/attack_alien(mob/living/user as mob)
+	if(islarva(user))
+		return
+	attack_generic(user, 15)
+
+
+/obj/machinery/fishtank/attack_hand(mob/user as mob)
+
+	if(user.a_intent == I_HURT)
+		playsound(get_turf(src), 'sound/effects/glassknock.ogg', 80, 1)
+		user.visible_message("<span class='danger'>[user.name] bangs against the [src]!</span>", \
+							"<span class='danger'>You bang against the [src]!</span>", \
+							"You hear a banging sound.")
+	else
+		playsound(src.loc, 'sound/effects/glassknock.ogg', 80, 1)
+		user.visible_message("[user] taps on the [src].", \
+							"You tap on the [src].", \
+							"You hear a knocking sound.")
+
+
+/obj/machinery/fishtank/proc/hit(var/damage, var/mob/user)
+	cur_health = max(0, cur_health - damage)
+	check_health()
+	user.delayNextAttack(0.3 SECONDS)
+
+/obj/machinery/fishtank/proc/attack_generic(mob/living/user as mob, damage = 0)	//used by attack_alien, attack_animal, and attack_slime
+	cur_health = max(0, cur_health - damage)
+	if(cur_health <= 0)
+		user.visible_message("<span class='danger'>\The [user] smashes through \the [src]!</span>")
+		destroy()
+	else	//for nicer text
+		user.visible_message("<span class='danger'>\The [user] smashes into \the [src]!</span>")
+		playsound(loc, 'sound/effects/Glasshit.ogg', 100, 1)
+		check_health()
+
+/obj/machinery/fishtank/attackby(var/obj/item/O, var/mob/user as mob)
+	//Welders repair damaged tanks on help intent, damage on all others
+	if(iswelder(O))
+		var/obj/item/weapon/weldingtool/W = O
+		if(user.a_intent == I_HELP)
+			if(W.isOn())
```

Comment:
Don't we use silicate sprayers for repairing glass?

Also you'll want to use the use proc or equivalent so it consumes some of the welder fuel/silicate when repairing

Reply:
We do, Paradise do not have silicate sprayers.